lci_strategic_plugin/src/lci_strategic_plugin.cpp

```diff
@@ -932,10 +792,23 @@ void LCIStrategicPlugin::planWhenAPPROACHING(const cav_srvs::PlanManeuversReques
 
   ROS_DEBUG_STREAM("SPEED PROFILE CASE:" << ts_params.case_num);
 
-  // CASE SELECTION END
+  /////////////  4. Safety Check against traffic signals and Final Maneuver Generation //////////////
+
+   double safe_distance_to_stop = pow(current_state.speed, 2)/(2 * max_comfort_decel_norm_);
+  ROS_DEBUG_STREAM("safe_distance_to_stop at max_comfort_decel:  " << safe_distance_to_stop << ", max_comfort_decel_norm_: " << max_comfort_decel_norm_);
+
+  double desired_distance_to_stop = pow(current_state.speed, 2)/(2 * max_comfort_decel_norm_ * config_.deceleration_fraction) + config_.desired_distance_to_stop_buffer;
+  ROS_DEBUG_STREAM("desired_distance_to_stop at: " << desired_distance_to_stop << ", where effective deceleration rate is: " << max_comfort_decel_norm_ * config_.deceleration_fraction);
+  
+  desired_distance_to_stop = std::max(desired_distance_to_stop, config_.stopping_location_buffer);
+
+  ROS_DEBUG_STREAM("new desired_distance_to_stop: " << desired_distance_to_stop);
+
+  ROS_DEBUG_STREAM("distance_remaining_to_traffic_light:  " << distance_remaining_to_traffic_light << ", current_state.speed: " << current_state.speed);
 
   // Although algorithm determines nearest_green_time is possible, check if the vehicle can arrive with certainty (Case 1-7)
-  if (ts_params.is_algorithm_successful && ts_params.case_num != TSCase::CASE_8) 
+  if (ts_params.is_algorithm_successful && ts_params.case_num != TSCase::CASE_8 && 
+    (distance_remaining_to_traffic_light >= desired_distance_to_stop || !in_tdb)) // Given ET is in TDB, but vehicle is too close to intersection
```

Comment:
Am I misunderstanding? Comment seems inaccurate, code reads `|| !in_tdb` but comment says ET is TBD.

Reply:
Comment was incorrect due to copy/paste and not editing. I changed it